pyatlan/model/typedef.py

```diff
@@ -1,241 +1,462 @@
-        None, description="", alias="indexTypeESFields"
+        description="Internal use only.", alias="indexTypeESFields"
     )
-    is_default_value_null: Optional[bool] = Field(None, description="TBC")
+    is_default_value_null: Optional[bool] = Field(description="TBC")
+
+    @staticmethod
+    def create(
+        display_name: str,
+        attribute_type: AtlanCustomAttributePrimitiveType,
+        multi_valued: bool = False,
+        options_name: Optional[str] = None,
+    ) -> AttributeDef:
+        from pyatlan.model.assets import validate_required_fields
+
+        validate_required_fields(
+            ["display_name", "attribute_type"],
+            [display_name, attribute_type],
+        )
+        # Explicitly set all defaults to ensure inclusion during pydantic serialization
+        builder = AttributeDef(
+            display_name=display_name,
+            options=AttributeDef.Options.create(
+                attribute_type=attribute_type, options_name=options_name
+            ),
+            is_new=True,
+            cardinality=Cardinality.SINGLE,
+            description="",
+            name="",
+            include_in_notification=False,
+            is_indexable=True,
+            is_optional=True,
+            is_unique=False,
+            values_min_count=0,
+            values_max_count=1,
+        )
+        add_enum_values = attribute_type == AtlanCustomAttributePrimitiveType.OPTIONS
+        if attribute_type == AtlanCustomAttributePrimitiveType.OPTIONS:
+            base_type = options_name
+        elif attribute_type in (
+            AtlanCustomAttributePrimitiveType.USERS,
+            AtlanCustomAttributePrimitiveType.GROUPS,
+            AtlanCustomAttributePrimitiveType.URL,
+            AtlanCustomAttributePrimitiveType.SQL,
+        ):
+            base_type = AtlanCustomAttributePrimitiveType.STRING.value
+        else:
+            base_type = attribute_type.value
+        if multi_valued:
+            builder.type_name = "array<" + str(base_type) + ">"
+            builder.options.multi_value_select = True
+        else:
+            builder.type_name = base_type
+        if add_enum_values:
+            from pyatlan.cache.enum_cache import EnumCache
+
+            if enum_def := EnumCache.get_by_name(str(options_name)):
+                builder.enum_values = enum_def.get_valid_values()
+            else:
+                builder.enum_values = []
+        return builder
+
+    def is_archived(self) -> bool:
+        if opt := self.options:
+            return bool(opt.is_archived)
+        else:
+            return False
+
+    def archive(self, by: str) -> AttributeDef:
+        if self.options:
+            removal_epoch = int(time.time() * 1000)
+            self.options.is_archived = True
+            self.options.archived_by = by
+            self.options.archived_at = removal_epoch
+            self.display_name = self.display_name + "-archived-" + str(removal_epoch)
```

Comment:
I would suggest to use of an f-string as follows:
`self.display_name = f"{self.display_name}-archived-{removal_epoch}"`